homeassistant/components/devolo_home_control/sensor.py

```diff
@@ -0,0 +1,92 @@
+"""Platform for sensor integration."""
+import logging
+
+from homeassistant.components.sensor import (
+    DEVICE_CLASS_HUMIDITY,
+    DEVICE_CLASS_ILLUMINANCE,
+    DEVICE_CLASS_TEMPERATURE,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import DOMAIN
+from .devolo_device import DevoloDeviceEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Get all sensor devices and setup them via config entry."""
+    entities = []
+
+    for device in hass.data[DOMAIN]["homecontrol"].multi_level_sensor_devices:
+        for multi_level_sensor in device.multi_level_sensor_property:
+            entities.append(
+                DevoloMultiLevelDeviceEntity(
+                    homecontrol=hass.data[DOMAIN]["homecontrol"],
+                    device_instance=device,
+                    element_uid=multi_level_sensor,
+                )
+            )
+    async_add_entities(entities, False)
+
+
+class DevoloMultiLevelDeviceEntity(DevoloDeviceEntity):
+    """Representation o a multi level sensor within devolo Home Control."""
+
+    def __init__(self, homecontrol, device_instance, element_uid):
+        """Initialize a devolo multi level sensor."""
+        super().__init__(
+            homecontrol=homecontrol,
+            device_instance=device_instance,
+            element_uid=element_uid,
+            name=f"{device_instance.itemName} {device_instance.multi_level_sensor_property.get(element_uid).sensor_type}",
```

Comment:
If you call the `super()` later, you have most of these values resolved already

Reply:
Done in 2ea2a8741e35aa41beeb931226b86f584a60b3bf